packages/common-all/src/parse.ts

```diff
@@ -0,0 +1,40 @@
+import * as v from "@badrap/valita";
```

Comment:
any reason we don't just stick with zod since we already use it in the publishing side?

Reply:
i had some issues with self-referencing types and using [Discriminated unions](https://github.com/colinhacks/zod#discriminated-unions) and valita tries (does not) to align with zod's api and says its faster and smaller. but I think it would be good to stay with zod. I will find a solution to revert this back to zod.

Reply:
parse doesn't have to be perfect. we can even have custom js logic in here or duplicate code. the tradeoff here is having a simpler mental model when working with the code base as a whole

Reply:
moved back to `zod` [`7fbea10` (#3448)](https://github.com/dendronhq/dendron/pull/3448/commits/7fbea10baca9e9f5c2da771a0e4cc989bbfef078)